src/app/src/main/java/com/example/bookmark/MyBooksActivity.java

```diff
@@ -1,18 +1,127 @@
 package com.example.bookmark;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.content.Intent;
 import android.os.Bundle;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ListView;
+
+import com.example.bookmark.adapters.BookList;
+import com.example.bookmark.models.Book;
+import com.example.bookmark.models.Owner;
+import com.example.bookmark.models.Request;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.material.floatingactionbutton.FloatingActionButton;
+
+import java.util.ArrayList;
 
 /**
- * TODO: Description of class.
+ * This activity shows a user a list of their books.
+ * They can select a book to see and edit the details of a book.
+ * They can also add a book from here
+ *
+ *  TODO: Add more to these? Classes/Listeners?
  * @author Mitch Adam.
  */
 public class MyBooksActivity extends AppCompatActivity {
+    // TODO: Figure out the back navigation
+    // Going to need some sort of owner or uid
+
+    ArrayList<Book> allBooks = new ArrayList<Book>();;
+    ArrayList<Book> filteredBooks;
```

Comment:
A good habit in Java is to prefer the List interface to ArrayList for the declaration so it's easy to swap out the type later. Also an extra semicolon snuck in.
```suggestion
    List<Book> allBooks = new ArrayList<Book>();
    List<Book> filteredBooks;
```